### ISSUE TITLE
Rename `ref` to `domRef`

### ISSUE THREAD

**Issue by appsforartists** [smyte/jsxstyle#84]
In the absence of `jsxstyle`, `<Row/>` would look like this:

```jsx
<div
  style = {
    {
      display: 'flex',
      flexDirection: 'row',
    }
  }
/>
```

and if you needed access to the underlying DOM element, you could use the `ref` prop:

```jsx
<div
  ref = { this.useDOMElement }
  style = {
    {
      display: 'flex',
      flexDirection: 'row',
    }
  }
/>
```

Unfortunately, `ref` is a magic prop that the `Row` component can't access.  Used on `Row` it will receive a reference to the React component instance, not the element.  [The React docs](https://reactjs.org/docs/refs-and-the-dom.html#exposing-dom-refs-to-parent-components) recommend using a separate prop for this and passing it through manually.

The equivalent `Row` code, after this change, would be:

```jsx
<Row
  domRef = { this.useDOMElement }
/>
```

**Comment by meyer** [comment]
Makes sense to me. This should be a pretty straightforward change, at least on the runtime side of things. If you feel like digging into the code, feel free; otherwise I’ll take a crack at this in the next couple of days.

Things will get a bit hairy with `jsxstyle-loader`, since it swaps `jsxstyle` components with the `component` prop value (or a `div`) if all styles are extracted. Shouldn’t be too wild, but `extractStyles.js` is a thousand lines of somewhat unfriendly AST parsing code. Should probably be modularised and better documented so other people can contribute 😬

**Comment by meyer** [comment]
I was thinking about this change a bit more as it relates to `jsxstyle-loader`. What makes the most sense to me: if `jsxstyle-loader` sees a `ref` prop specified on a `jsxstyle` component, it will treat the prop as dynamic. This will prevent the component from being replaced with a `div` or the user-specified `component`. `domRef` will be passed through as `ref` to the child component if all style props are static, otherwise it’ll be left alone.

**Comment by appsforartists** [comment]
Does that mean this is 👍 to merge?

**Comment by meyer** [comment]
@appsforartists it needs tests and `jsxstyle-loader` support but I can take care of that.

**Comment by appsforartists** [comment]
Thanks!

**Comment by meyer** [comment]
@appsforartists I realised in the shower this morning that I’m probably overthinking this. I think this can be resolved by removing the (nonfunctional and incorrect) references to `ref`-passing from runtime and the README.

You can already pass the ref to the underlying component by using the `props` prop:

```jsx
<Block
  props={{ ref: this.attachRef }}
  ref={this.attachComponentRefForSomeReason}
/>
```

Thoughts?

**Comment by meyer** [comment]
I’m also thinking `jsxstyle-loader` should warn if the `ref` prop is present on a jsxstyle component, since it just adds a layer of indirection and it makes the component unable to be completely extracted.

**Comment by appsforartists** [comment]
Good thinking!  I forgot there's a `props` prop now.

**Comment by meyer** [comment]
@appsforartists I just published your two changes as `jsxstyle@2.0.1`. Thanks again! 🎉🎉🎉
